src/runtime/cuda/cuda_code_object.cpp

```diff
@@ -87,14 +87,30 @@ result build_cuda_module_from_ptx(CUmod_st *&module, int device,
   // context on that device. This is important for the subsequent driver
   // API calls which assume that CUDA context has been created.
   cudaFree(0);
-  
+
+  static constexpr std::size_t num_options = 2;
+  std::array<CUjit_option, num_options> option_names{};
+  std::array<void*, num_options> option_vals{};
+
+  // set up size of compilation log buffer
+  option_names[0] = CU_JIT_ERROR_LOG_BUFFER_SIZE_BYTES;
+  static constexpr std::size_t error_log_buffer_size = 1024*1024;
```

Comment:
[question] Not sure on the best number here?  Another idea, is it worth making the log buffer `thread_local`? - then the size is less important.

Reply:
Probably it's not worth making it thread-local. JIT, when it happens, is always going to be expensive.
I would suggest though to reduce the size a bit. In case of an error, I'm skeptical that people would find it helpful to read through 1M of error messages :-) My guess is that some KB are enough to be helpful.

Reply:
:+1: reduced to 10KiB in 9ce5ec94